src/shared/components/formik-fields/__tests__/DropdownField.spec.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { configure, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { Formik, FormikConfig } from 'formik';
+import { DropdownItemObject } from '../../dropdown/BasicDropdown';
+import DropdownField from '../DropdownField';
+import '@testing-library/jest-dom';
+
+type FormData = {
+  dropdownValue: string;
+};
+const Wrapper: React.FC<FormikConfig<FormData>> = ({ children, ...formikConfig }) => (
+  <Formik {...formikConfig}>
+    {(formikProps) => (
+      <form onSubmit={formikProps.handleSubmit}>
+        {children}
+        <input type="submit" value="Submit" />
+      </form>
+    )}
+  </Formik>
+);
+
+const items: DropdownItemObject[] = [
+  { key: 'key1', value: 'value1' },
+  { key: 'key2', value: 'value2' },
+  { key: 'key3', value: 'value3' },
+];
+
+const fieldName: string = 'dropdownValue';
+
+const initialValues = {
+  dropdownValue: items[0].value,
+};
+
+configure({ testIdAttribute: 'data-test' });
+
+afterEach(jest.resetAllMocks);
+
+describe('DropdownField', () => {
+  it('renders dropdown field', () => {
+    render(
+      <Wrapper initialValues={initialValues} onSubmit={jest.fn()}>
+        <DropdownField name={fieldName} label="label" helpText="helpText" items={items} />
+      </Wrapper>,
+    );
+    expect(screen.getByText('label')).toBeInTheDocument();
+    expect(screen.getByText('helpText')).toBeInTheDocument();
+    expect(screen.getByText(initialValues.dropdownValue)).toBeInTheDocument();
+  });
+
+  it('renders dropdown menu with all menu items on toggle button click', async () => {
+    render(
+      <Wrapper initialValues={initialValues} onSubmit={jest.fn()}>
+        <DropdownField name={fieldName} label="label" helpText="helpText" items={items} />
+      </Wrapper>,
+    );
+    expect(screen.queryByRole('menu')).toBeFalsy();
+    fireEvent.click(screen.getByText(initialValues.dropdownValue));
+    await waitFor(() => {
+      expect(screen.getByRole('menu')).toBeInTheDocument();
+      const menuItems = screen.getAllByRole('menuitem');
+      const isItemExist = (item) => menuItems.some((mi) => mi.nodeValue === item.value);
+      expect(items.every(isItemExist));
```

Comment:
Are we checking that `nodeValue` exists for all items in the menu? This expect call seems incomplete.
```suggestion
      expect(items.every(isItemExist)).toBeTruthy();
```